homeassistant/core.py

```diff
@@ -241,9 +241,24 @@ class HomeAssistant:
     auth: AuthManager
     http: HomeAssistantHTTP = None  # type: ignore[assignment]
     config_entries: ConfigEntries = None  # type: ignore[assignment]
+    _initialized: bool = False
+    _instance: HomeAssistant | None = None
```

Comment:
As an alternative, we could rename `__init__` and call it directly from `__new__`.
Pylint doesn't support adding attributes in `__new__` though, so it means we have to add explicit type hints for all instance attributes too: https://github.com/microsoft/pylance-release/issues/573